src/main/scala/lms/thirdparty/cuda/cuda.scala

```diff
@@ -1093,6 +1093,55 @@ trait CudaLibs extends CudaOps {
         }
       }
   }
+
+  def cudaSplit3D0[N:Numeric:Manifest](implicit __pos: SourceContext) = cudaGlobalFun {
+    (in: Rep[Array[N]], out1: Rep[Array[N]], out2: Rep[Array[N]], d0: Rep[Int], split: Rep[Int], input_size: Rep[Int]) => {
+      generate_comment("this is cuda split kernel. It takes a 3D array and splits on the innermost dimension (dim2).")
+      generate_comment("in: input array")
+      generate_comment("out1: first output array")
+      generate_comment("out2: second output array")
+      generate_comment("d0: dim2 of the input array")
+      generate_comment("split: dim2 of the first output array")
+      generate_comment("call constraint: 0 < split < in.dim2 ")
+      generate_comment("call constraint: out1.dim2 + out2.dim2 = in.dim2 ")
```

Comment:
I think using `out1` or `d0` in the `generate_comments` is not ideal. If you check the generated code, the parameters do not keep the scala name. So I think the best we can do is to use `arg0` `arg1` ..

In the generated comments, I think one important comment is how the kernel is called, i.e., what is the assumed `dim3(x,y,z)`.

Reply:
fixed in 1f17da3